keps/2937-resource-transformer/README.md

```diff
@@ -0,0 +1,560 @@
+      - name: nvidia.com/mig-1g.5gb
+        nominalQuota: 40  # 200G/5g
+      - name: nvidia.com/mig-2g.10gb
+        nominalQuota: 20  # 200G/10g
+      - name: nvidia.com/mig-4g.20gb
+        nominalQuota: 4  # less than the 10 that would be allowed by the acelerator-memory quota
+      - name: nvidia.com/mig-7g.40gb
+        nominalQuota: 0  # not allowed -- inefficient use of GPU resources
+```
+
+[mig]: https://docs.nvidia.com/datacenter/cloud-native/kubernetes/latest/index.html#the-mixed-strategy
+[instaslice]: https://github.com/openshift/instaslice-operator
+
+#### Story 2
+
+This KEP would allow Kueue to augment a Job's resource requests
+for quota purposes with additional resources that are not actually 
+available in the Nodes of the cluster. For example, it could be used to 
+define quotas in terms of a fictional currency `example.com/credits`
+to give "costs" to various resources. This could be used to approximate
+the actual monetary cost of a cloud resource (for example a larger or more powerful GPU
+could cost twice as many `credits` as a smaller or previous generation GPU).
+We'd expect in this usage pattern that `Replace` would be used so that
+resource list is just augmented with the computed `credits` resource.  We'd also
+expect that a ClusterQueue's `borrowingLimit` for `credits` would be set to `0`
+to enforce the intended fiscal constraint.
+
+## Design Details
+
+Since Kueue's workload.Info struct is already a well-defined
+place where resource filtering is being applied to compute the effective resource
+request from a Workload's PodSpec, the core of the implementation is
+straightforward and follows from how `excludedResourcePrefixes` are already processed.
+
+The non-trivial design issues are how to configure the transformations,
+the possible granularity of that configuration, and how to maximize observability
+of the effective resources requested by a Workload.
+
+### Specifying the Transformation
+
+A transformation matrix is defined as a single object.  Most likely
+this object would be embedded inside Kueue's `Configuration` API, but
+it would also be possible to make it a separate top-level config API or 
+use a config map containing a single yaml field that would be deserialized
+into the type defined below.  To support finer-grained configuration, instead
+of defining a single controller-wide transformation matrix, they could be embedded
+in ClusterQueue ResourceGroups (perhaps implicitly by instead embedding the
+transformation matrix in a ResourceFlavor).
+
+```go
+type ResourceTransformationStrategy string
+const Retain ResourceTransformationStrategy = "Retain"
+const Replace ResourceTransformationStrategy = "Replace"
+
+type ResourceTransformation struct {
+  // key is the output resource name and value is the quantity of the output resource 
+  // that corresponds to 1 unit of the input resource
+  ConversionBaseValues map[corev1.ResourceName]resource.Quantity `json:"conversionBaseValues"`
```

Comment:
```suggestion
  ConversionBaseValues corev1.ResourceList `json:"conversionBaseValues"`
```

Is there any motivation to select the typed "map[corev1.ResourceName]resource.Quantity" instead of the "corev1.ResourceList"?

Reply:
simplified as suggested.  Was the result of several steps of evolution.